odoo/service/server.py

```diff
@@ -332,6 +333,18 @@ def close_socket(self, sock):
                 raise
         sock.close()
 
+    def atexit(self, func):
+        """ Register a cleanup function to be executed when the server stops """
+        self._atexit_funcs.append(func)
+
+    def _run_atexit_funcs(self):
```

Comment:
Why don't you call this one `stop()`, and use `super().stop()` in children classes?

Reply:
The code execution is dependent of the context (graceful shutdown vs force shutdown) so I think it is a better design to have an explicit method than a generic `stop()` that is sometime called and sometime not.

Reply:
BUT such branching is specific to the prefork server (which has a bad design), so let's go with a super `stop()`